use-cases/school-timetabling/src/main/java/org/acme/schooltimetabling/service/TimeTableDemoResource.java

```diff
@@ -8,31 +8,58 @@
 import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
 import org.acme.schooltimetabling.domain.Lesson;
 import org.acme.schooltimetabling.domain.Room;
 import org.acme.schooltimetabling.domain.TimeTable;
 import org.acme.schooltimetabling.domain.Timeslot;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
 
-@Path("demo")
+@Tag(name = "Demo data sets", description = "Timefold-provided demo school time table data sets")
+@Path("demo/datasets")
 public class TimeTableDemoResource {
 
-    public enum DemoDataType {
+    public enum DemoDataSet {
         SMALL,
         LARGE
     }
 
-    // TODO openapi
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "List of demo data sets represented as IDs",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = DemoDataSet.class, type = SchemaType.ARRAY))) })
+    @Operation(summary = "List demo data sets")
     @GET
-    @Path("data-types")
-    public DemoDataType[] list() {
-        return DemoDataType.values();
+    @Path("ids")
+    public DemoDataSet[] list() {
+        return DemoDataSet.values();
     }
 
-    // TODO openapi
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Unsolved demo time table",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = TimeTable.class))),
+            @APIResponse(responseCode = "404", description = "Demo data set does not exist",
+                    content = @Content(mediaType = MediaType.TEXT_PLAIN))})
+    @Operation(summary = "Find an unsolved demo time table by ID")
     @GET
-    @Path("data/{demoDataType}")
-    public TimeTable generate(@PathParam("demoDataType") DemoDataType demoDataType) {
-        String name = demoDataType.name();
+    @Path("{dataSetId}")
+    public Response generate(@Parameter(description = "Unique identifier of the demo data set",
+            required = true) @PathParam("dataSetId") String dataSetId) {
+        DemoDataSet demoDataSet;
+        try {
+            demoDataSet = DemoDataSet.valueOf(dataSetId);
+        } catch (IllegalArgumentException illegalArgumentException) {
+            return Response.status(Response.Status.NOT_FOUND).entity("Demo data set " + dataSetId + " not found.").build();
```

Comment:
In the quickstarts, I wouldn't bother with Response objects. It just makes it more complex.

Just return TimeTable type and throw an IllegalArgumentException. Let the default exception handlers turn it into a 500 (which is good enough).

=> Less code to grok for users.

Reply:
Reworked to use exception mappers as agreed.